### ISSUE TITLE
Re-expose customEndpoint configuration option

### ISSUE THREAD

**Issue by jeantil** [googleapis/nodejs-storage#1315]
**Is your feature request related to a problem? Please describe.**

I am trying to write tests for code which uses GCS. these tests are run on a CI platform. Having the tests run against a real bucket means potential erratic failures because of network issues and possibly billing for "useless" operations. I am trying to use  a locally deployable emulator (even if it doesn't support all the features) but that proves harder than it should be. 

In https://github.com/googleapis/nodejs-storage/issues/18#issuecomment-354029129 @stephenplusplus suggested to use 
```
const Storage = require('@google-cloud/storage')
const gcs = new Storage({ customEndpoint: true })
```
but this feature no longer  works, even though the feature is still available in the [`@googlecloud/common`](https://github.com/googleapis/nodejs-common/blob/ab509533c8aac045d46cf1a999f0ae37633767c1/src/util.ts#L495) library

 **Describe the solution you'd like**

I would like the at least that parameter to be restored  in  `StorageOptions` and propagated to the GoogleAuth client. I am not sure if `StorageOption` should add more options from `MakeAuthenticatedRequestFactoryConfig` 

The customEndpoint parameter then needs to propagated when building the config object which is passed to the `Service` abstract class in the `Storage` class constructor

 **Describe alternatives you've considered**

My current workaround is to use the `apiEndpoint` to run against a local store but provide valid credentials to a free firebase project in order to pass through auth. This means our test suite cannot work without access to google's servers and the CI plaform uselessly sends requests to google auth servers.
 
I am willing to contribute a PR if it can help.

**Comment by jeantil** [comment]
thanks for the quick update !

**Comment by rdlowrey** [comment]
Impacted me as well -- thanks for fixing 👍

**Comment by stephenplusplus** [comment]
Thanks for your help and patience! It's released now under 5.4.0.
